bundles/org.openhab.core/src/main/java/org/openhab/core/internal/items/ItemRegistryImpl.java

```diff
@@ -280,6 +281,17 @@ protected void unsetEventPublisher(EventPublisher eventPublisher) {
         }
     }
 
+    @Override
+    @Reference
+    protected void setReadyService(ReadyService readyService) {
+        super.setReadyService(readyService);
+    }
+
+    @Override
+    protected void unsetReadyService(ReadyService readyService) {
+        super.unsetReadyService(readyService);
+    }
```

Comment:
Did you also look into using constructor injection for injecting the `ReadyService` into the registries?

Reply:
No, I didn't want to mess with that in that PR as it can easily be a bigger refactoring. That's imho rather something for after the release.

Reply:
I can imagine that, but if you had already given it a try and it didn't work out well... then your insights on this might help if someone else looks into it one day. :slightly_smiling_face: